frontend/app/dashboard/page.tsx

```diff
@@ -1,104 +1,163 @@
-import { EventData } from "@/interfaces/EventTypes";
-import { getAllEvents } from "@/services/eventsService";
-import { useRouter, useSearchParams } from "next/navigation";
+import { getAllEvents, getEventById } from "@/services/eventsService";
 import { useEffect, useState } from "react";
+import { searchEventsByKeyword } from "@/services/eventsService";
+import { useRouter } from "next/navigation";
+import { useSearchParams } from "next/navigation";
+import { EventData } from "@/interfaces/EventTypes";
+import Loading from "@/components/Loading";
+import FilterBanner from "@/components/Filter/FilterBanner";
+import EventCard from "@/components/EventCard";
 import { Alert } from "@material-tailwind/react";
 
 export default function Dashboard() {
-  const [loading, setLoading] = useState(true);
-  const [allEventsDataList, setAllEventsDataList] = useState<EventData[]>([]);
-  const [eventDataList, setEventDataList] = useState<EventData[]>([]);
-  const [showLoginSuccess, setShowLoginSuccess] = useState(false);
-  const router = useRouter();
-  const searchParams = useSearchParams();
-  useEffect(() => {
-    getAllEvents()
-      .then((events) => {
-        setEventDataList(events);
-        setAllEventsDataList(events);
-      })
-      .finally(() => {
-        setLoading(false);
-      });
-  }, []);
+    const [loading, setLoading] = useState(true);
+    const [allEventsDataList, setAllEventsDataList] = useState<EventData[]>([]);
+    const [eventDataList, setEventDataList] = useState<EventData[]>([]);
+    const [searchDataList, setSearchDataList] = useState<EventData[]>([]);
+    const [showLoginSuccess, setShowLoginSuccess] = useState(false);
+    const router = useRouter();
+    const [currentUrl, setCurrentUrl] = useState(
+        typeof window !== "undefined" ? window.location.href : ""
+    );
+    const searchParams = useSearchParams();
+    const [srcLocation, setSrcLocation] = useState<string>("");
+    // useEffect(() => {
+    //   getAllEvents()
+    //     .then((events) => {
+    //       setEventDataList(events);
+    //       setAllEventsDataList(events);
+    //     })
+    //     .finally(() => {
+    //       setLoading(false);
+    //     });
+    // }, []);
+    const getQueryParams = () => {
+        if (typeof window === "undefined") {
+            // Return some default or empty values when not in a browser environment
+            return { event: "", location: "" };
+        }
+        const searchParams = new URLSearchParams(window.location.search);
+        setSrcLocation(searchParams.get("location") || "");
+        return {
+            event: searchParams.get("event") || "",
+            location: searchParams.get("location") || "",
+        };
+    };
 
-  useEffect(() => {
-    const login = searchParams?.get("login");
-    if (login === "success") {
-      setShowLoginSuccess(true);
+    useEffect(() => {
+        setLoading(true);
+        console.log("test");
```

Comment:
remove console log

Reply:
ensure to remove other unnecessary/testing/dev console logs throughout the changed files